### PR TITLE
Added Azure.Storage.DefenderCloud.SensitiveData

### PR DESCRIPTION
**Pull request by BenjaminEngeset**
## PR Summary

Part of #2207 

Added Azure.Storage.DefenderCloud.SensitiveData rule.

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [x] Unit tests created/ updated
  - [x] Rule documentation created/ updated
  - [x] Link to a filed issue
  - [x] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

**Comment by BenjaminEngeset** [comment]
@BernieWhite When you have the time.

File: docs/en/rules/Azure.Storage.DefenderCloud.SensitiveData.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+- Detect possible security incidents resulting is data exposure.
+
+When enabling sensitive data threat detection, the sensitive data categories include built-in sensitive information types (SITs) in the default list of Microsoft Purview. 
+It is possible to customize the Data Sensitivity Discovery for a organization, by creating custom sensitive information types (SITs).
+
+Sensitive data threat detection in Microsoft Defender for Storage can be enabled at the resource level. However, the general recommandation is to enable it at the subscription level and by doing so ensures all storage accounts in the subscription will be protected, including future ones. Defender for Storage settings on each storage account is inherited by the subscription level settings.
+
+It is also worth to mention that the resouce level enablement can be useful when:
+
+- Override subscription level settings to configure specific storage accounts with custom sensitive data threat detection settings that differ from the settings configured at the subscription level.
```
Comment (BernieWhite) [inline docs/en/rules/Azure.Storage.DefenderCloud.SensitiveData.md:32]:
```suggestion
When enabling sensitive data threat detection, the sensitive data categories include built-in sensitive information types (SITs) in the default list of Microsoft Purview.
It is possible to customize the Data Sensitivity Discovery for a organization, by creating custom sensitive information types (SITs).

Sensitive data threat detection in Microsoft Defender for Storage can be enabled at the subscription level and by doing so ensures all storage accounts in the subscription will be protected, including future ones.

When overriding sensitive data threat detection on individual Storage Account it is possible to configure custom sensitive data threat detection settings that differ from the settings configured at the subscription level.
```

**Review by BernieWhite — CHANGES_REQUESTED** [review]
Great work @BenjaminEngeset. I have added a few suggestion for the documentation.

**Comment by BenjaminEngeset** [comment]
@BernieWhite Thanks for the great suggestions, they have been implemented. We had an overlapping rule reference, so I had to change it to `000391`.

**Review by BernieWhite — APPROVED** [review]
Great work @BenjaminEngeset. All good to merge.

**Pull request merged**